### PR TITLE
Enso Diags regression coefficient

### PR DESCRIPTION
**Pull request by forsyth2**
This is the third iteration pull request for ENSO Diags - the first two being #254 and #271. This pull request implements the first component of ENSO diagnostics - regression of a variable over Nino region SST anomalies.

### CODE REVIEW

**Comment by forsyth2** [comment]
@chengzhuzhang I still want to self-review this pull request but feel free to try running code from this branch.
- All unit tests pass. I added a simple `unit_tests.sh` to run all tests. (I saw `test.sh` but that didn't successfully run everything...)
- I think I have the contour algorithm working for small and/or non-symmetric values, so you might want to try that out.
- I moved the colorbar down so if the multiplier (e.g., "1e-13") is above it, then it won't overlap the stats.

File: acme_diags/driver/enso_diags_driver.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+import genutil
+import acme_diags
+from acme_diags.derivations import acme, default_regions
+from acme_diags.driver import utils
+from acme_diags.metrics import rmse, corr, min_cdms, max_cdms, mean, std
+from acme_diags.plot.cartopy.enso_diags_plot import plot
+
+def calculate_nino_index(nino_region_str, parameter):
+    """
+    Use the built-in HadISST nino index time series from  http://www.esrl.noaa.gov/psd/gcos_wgsp/Timeseries/
```
Comment (forsyth2) [inline acme_diags/driver/enso_diags_driver.py]: Extra space after "from"

File: acme_diags/driver/enso_diags_driver.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                ref_domain, ref_reg_coe, ref_confidence_levels = perform_regression(ref_data, parameter, var, region, land_frac, ocean_frac, ref_nino_index, nino_region_str)
+
+                # Difference
+                # Regrid towards the lower resolution of the two variables for calculating the difference.
+                test_reg_coe_regrid, ref_reg_coe_regrid = utils.general.regrid_to_lower_res(test_reg_coe, ref_reg_coe, parameter.regrid_tool, parameter.regrid_method)
+                diff = test_reg_coe_regrid - ref_reg_coe_regrid
+
+                # Metrics
+                metrics_dict = create_metrics(ref_reg_coe, test_reg_coe, ref_reg_coe_regrid, test_reg_coe_regrid, diff)
+                # If nor defined, determined contour_levels
```
Comment (forsyth2) [inline acme_diags/driver/enso_diags_driver.py]: "determine"

File: acme_diags/driver/enso_diags_driver.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                        metrics_dict['test']['min']
+                    ))
+                    max_contour_level = math.ceil(max(
+                        metrics_dict['ref']['max'],
+                        metrics_dict['test']['max']
+		    ))
+                    CENTER_ON_ZERO = True
+                    if CENTER_ON_ZERO:
+                        bound = max(abs(max_contour_level), abs(min_contour_level))
+                        lower_bound = bound
```
Comment (forsyth2) [inline acme_diags/driver/enso_diags_driver.py]: `lower_bound = -bound`

File: acme_diags/driver/enso_diags_driver.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                    else:
+                        lower_bound = min_contour_level
+                        upper_bound = max_contour_level
+                        contour_level_range = max_contour_level - min_contour_level
+                    step_size = (contour_level_range / 10)
+                    if step_size > 1:
+                        step_size = int(step_size)
+                    elif step_size == 0:
+                        step_size = 1/10
+                    contour_levels = list(numpy.arange(-lower_bound, upper_bound + 1, step_size))
```
Comment (forsyth2) [inline acme_diags/driver/enso_diags_driver.py]: `lower_bound`, no negative

File: acme_diags/plot/cartopy/enso_diags_plot.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+
+def plot(reference, test, diff, metrics_dict, ref_confidence_levels, test_confidence_levels, parameter):
+    if parameter.backend not in ['cartopy', 'mpl', 'matplotlib']:
+        return
+
+    # Create figure, projection
+    fig = plt.figure(figsize=parameter.figsize, dpi=parameter.dpi)
+    #proj = ccrs.PlateCarree(central_longitude=180)
+    proj = ccrs.PlateCarree(central_longitude=179.99)
```
Comment (forsyth2) [inline acme_diags/plot/cartopy/enso_diags_plot.py]: Maybe we should link to the GitHub issue that forces us to use this central longitude. @chengzhuzhang https://github.com/SciTools/cartopy/issues/946 is the correct one, right?

Comment (chengzhuzhang) [reply acme_diags/plot/cartopy/enso_diags_plot.py]: Good idea! Yes it's the correct issue to refer to.

**Pull request merged**